packages/foundry/src/vault/adapter/aave/aaveV2/AaveV2Adapter.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: GPL-3.0
+// Docgen-SOLC: 0.8.15
+
+pragma solidity ^0.8.15;
+
+import { AdapterBase, IERC20, IERC20Metadata, SafeERC20, ERC20, Math, IStrategy, IAdapter } from "../../abstracts/AdapterBase.sol";
+import { WithRewards, IWithRewards } from "../../abstracts/WithRewards.sol";
+import { ILendingPool, IAaveMining, IAToken } from "./IAaveV2.sol";
+
+/**
+ * @title   AaveV2 Adapter
+ * @author  amatureApe
+ * @notice  ERC4626 wrapper for AaveV2 Vaults.
+ *
+ * An ERC4626 compliant Wrapper for https://github.com/aave/protocol-v2/blob/master/contracts/protocol/lendingpool/LendingPool.sol.
+ * Allows wrapping AaveV2 aTokens with or without an active Liquidity Mining.
+ * Allows for additional strategies to use rewardsToken in case of an active Liquidity Mining.
+ */
+
+contract AaveV2Adapter is AdapterBase, WithRewards {
+  using SafeERC20 for IERC20;
+  using Math for uint256;
+
+  string internal _name;
+  string internal _symbol;
+
+  // @notice The Aave aToken contract
+  IAToken public aToken;
+
+  // @notice The Aave liquidity mining contract
+  IAaveMining public aaveMining;
+
+  // @notice Check to see if Aave liquidity mining is active
+  bool public isActiveMining;
+
+  // @notice The address that will receive the liquidity mining rewards if LM is active
+  address public rewardsRecipient;
```

Comment:
I wasn't 100% sure how you'd prefer the rewardsRecipient so I just left it to be passed via initialization for now. I suppose this should be address(this) though?